Sources/BulletinViewController.swift

```diff
@@ -337,6 +338,58 @@ extension BulletinViewController: UIViewControllerTransitioningDelegate {
 
 }
 
+// MARK: - Keyboard
+
+extension BulletinViewController {
+    func setUpKeyboardLogic() {
+        NotificationCenter.default.addObserver(self, selector: #selector(onKeyboardShow), name: .UIKeyboardWillShow, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(onKeyboardHide), name: .UIKeyboardWillHide, object: nil)
+    }
+
+    @objc func onKeyboardShow(_ notification: Notification) {
+        if let userInfo = notification.userInfo {
+            if let keyboardFrameFinal = userInfo[UIKeyboardFrameEndUserInfoKey] as? CGRect,
+                let duration = userInfo[UIKeyboardAnimationDurationUserInfoKey] as? Double,
+                let curveInt = userInfo[UIKeyboardAnimationCurveUserInfoKey] as? Int {
+
+                var animationCurve = UIViewAnimationCurve(rawValue: curveInt)
+                if animationCurve == nil {
+                    animationCurve = UIViewAnimationCurve.linear
+                }
+
+                UIView.animate(withDuration: duration, delay: 0, options: animationCurve!.toOptions(), animations: {
+                    self.contentBottomConstraint.constant = -(keyboardFrameFinal.size.height + 12) // same value as in moveIntoPlace()
+                    self.contentView.superview?.layoutIfNeeded()
+                }, completion: nil)
+            }
+        }
+    }
+
+    @objc func onKeyboardHide(_ notification: Notification) {
+        if let userInfo = notification.userInfo {
+            if let duration = userInfo[UIKeyboardAnimationDurationUserInfoKey] as? Double,
+                let curveInt = userInfo[UIKeyboardAnimationCurveUserInfoKey] as? Int {
+
+                var animationCurve = UIViewAnimationCurve(rawValue: curveInt)
```

Comment:
You could use `nil` coalescing here:

~~~swift
let animationCurve = UIViewAnimationCurve(rawValue: curveInt) ?? .linear
~~~

This way, `animationCurve` stays constant and is not an optional.